k8s-bench/eval.go

```
@@ -0,0 +1,162 @@
+
+			for _, model := range config.Models[provider] {
+				success, err := evaluateTask(config, taskID, task, provider, model)
+				if err != nil {
+					fmt.Printf("Error evaluating task %s with %s/%s: %v\n", taskID, provider, model, err)
+					results[taskID][provider][model] = false
+					continue
+				}
+				results[taskID][provider][model] = success
+			}
+		}
+	}
+
+	printResults(results)
+	return nil
+}
+
+func loadTasks(config EvalConfig) (map[string]Task, error) {
+	tasks := make(map[string]Task)
+
+	entries, err := os.ReadDir(config.TasksDir)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		taskID := entry.Name()
+		if config.TaskPattern != "" && !strings.Contains(taskID, config.TaskPattern) {
+			continue
+		}
+
+		taskFile := filepath.Join(config.TasksDir, taskID, "task.yaml")
+
+		data, err := os.ReadFile(taskFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read task file %s: %w", taskFile, err)
+		}
+
+		var task Task
+		if err := yaml.Unmarshal(data, &task); err != nil {
+			return nil, fmt.Errorf("failed to parse task file %s: %w", taskFile, err)
+		}
+
+		// Skip disabled tasks
+		if task.Disabled {
+			fmt.Printf("Skipping disabled task: %s\n", taskID)
+			continue
+		}
+
+		tasks[taskID] = task
+	}
+
+	return tasks, nil
+}
+
+func evaluateTask(config EvalConfig, taskID string, task Task, provider, model string) (bool, error) {
+	// Run kuba
+	cmd := exec.Command(config.AgentBin,
+		"--kubeconfig", config.KubeConfig,
+		"--query", task.Goal,
+		"--llm-provider", provider,
+		"--model", model,
+	)
+
+	cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG=%s", config.KubeConfig))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	fmt.Printf("\nRunning kuba for task %s with %s/%s\n", taskID, provider, model)
+
+	if err := cmd.Run(); err != nil {
+		return false, fmt.Errorf("kuba failed: %w", err)
+	}
+
+	success := true
```

Nit: we might consider a string result, if there's no verifier I'm wary of calling it a "success" ?

---

+1, maybe just throw an error when there's no verifier